Cubical/Data/Int/Properties.agda

```diff
@@ -133,3 +100,156 @@ discreteInt (negsuc n) (negsuc m) with discreteℕ n m
 
 isSetInt : isSet Int
 isSetInt = Discrete→isSet discreteInt
+
+-- Compose sucPathInt with itself n times. Transporting along this
+-- will be addition, transporting with it backwards will be
+-- subtraction.
+
+addEq : ℕ → Int ≡ Int
+addEq zero = refl
+addEq (suc n) = compPath (addEq n) sucPathInt
+
+subEq : ℕ → Int ≡ Int
+subEq n i = addEq n (~ i)
+
+_+_ : Int → Int → Int
+m + pos n    = transport (addEq n) m
+m + negsuc n = transport (subEq (suc n)) m
+
+_-_ : Int → Int → Int
+m - pos zero    = m
+m - pos (suc n) = m + negsuc n
+m - negsuc n    = m + pos (suc n)
+
+-- We directly get that addition by a fixed number is an equivalence
+-- without having to do any induction!
+isEquivAddInt : (m : Int) → isEquiv (λ n → n + m)
+isEquivAddInt (pos n)    = isEquivTransport (addEq n)
+isEquivAddInt (negsuc n) = isEquivTransport (subEq (suc n))
+
+-- TODO: define multiplication by composing the addition equivalence
+-- with itself.
+
+private
+  +negsuc : ℕ → (Int → Int)
+  +negsuc 0 = predInt
+  +negsuc (suc n) z = predInt (+negsuc n z)
+
+  L0 : ∀ n z → +negsuc (suc n) z ≡ +negsuc n (predInt z)
+  L0 0 z = refl
+  L0 (suc n) z = cong predInt (L0 n z)
+
+  Lnegsuc : ∀ n z → z + negsuc n ≡ +negsuc n z
+  Lnegsuc 0 z = refl
+  Lnegsuc (suc n) z = compPath (Lnegsuc n (predInt z)) (sym (L0 n z))
+
+  L' : ∀ m n → m + (negsuc (suc n)) ≡ predInt (m + negsuc n)
+  L' m n = compPath (Lnegsuc (suc n) m) (cong predInt (sym (Lnegsuc n m)))
+
+  L3- : ∀ m n → m + (sucInt (negsuc n)) ≡ sucInt (m + (negsuc n))
+  L3- m 0 = sym (sucPred m)
+  L3- m (suc n) = compPath (sym (sucPred _)) (cong sucInt (sym (L' m n)))
+  
+  Lsuc : ∀ m n → m + (sucInt n) ≡ sucInt (m + n)
+  Lsuc m (pos n)    = refl
+  Lsuc m (negsuc n) = L3- m n
+
+  Lpred : ∀ m n → m + (predInt n) ≡ predInt (m + n)
```

Comment:
I think this and some of the above lemmas should be exported, e.g. Lnegsuc, L', L3- and Lsuc. (with good names inspired by the standard library names)

Reply:
What are some example of good names? I have a hard time come up with these. 

How about renaming `Lpred` to `+predInt≡predInt+`? This still have risk of misunderstanding. 
Maybe `m+[predIntn]≡predInt[m+n]`?